lib/pedant_mysql2.rb

```diff
@@ -46,7 +46,11 @@ def whitelist
     end
 
     def ignored?(warning)
-      whitelist.any? { |matcher| matcher =~ warning.message }
+      whitelist.any? { |matcher| matcher =~ warning.message } || drop_table_warning(warning)
+    end
+
+    def drop_table_warning(warning)
+      warning.query =~ /\ADROP TABLE IF EXISTS/ || warning.message =~ /\AUnknown table/
```

Comment:
Yeah, that is exactly the case. This gem is here to help people to find warning that can be actionable. This specific warning is not actionable, because this specific query was made to show a warning. You only use `IF EXISTS` in the `DROP TABLE` is you don't care if the table exists or not. If we raise an exception for this it means that the reason to use `IF EXISTS` is not a valid reason.